File: laboratorios/lab01/codigo/Arc.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package lab1;

/**
 *
 * @author Yhoan Alejandro Guzmán García
 * @author Juan Sebastián Pérez 
 */
public class Arc {
    private String nodeIDFrom;
    private String nodeIDTo;
    private int distance;
    private String name;
    public Arc(String nodeIDFrom, String nodeIDTo, int weight, String name) {
        this.nodeIDFrom = nodeIDFrom;
        this.nodeIDTo = nodeIDTo;
        this.distance = weight;
        this.name = name;
    }

    public String getNodeIDFrom() {
        return nodeIDFrom;
    }

    public void setNodeIDFrom(String nodeIDFrom) {
        this.nodeIDFrom = nodeIDFrom;
    }

    public String getNodeIDTo() {
        return nodeIDTo;
    }

    public void setNodeIDTo(String nodeIDTo) {
        this.nodeIDTo = nodeIDTo;
    }

    public int getDistance() {
        return distance;
    }

    public void setDistance(int distance) {
        this.distance = distance;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
    
    
}

File: proyecto/codigo/Arc.py
class Arc:
    """
        This class is the arc or edge object
        :author Yhoan Alejandro Guzmán García
        :author Juan Sebastián Pérez
    """

    def __init__(self, time, distance, nodeFrom, nodeTo):
        """
        Class constructor
        :param time: time between the nodes
        :param distance: distance between the nodes in meters
        :param nodeFrom: node object from
        :param nodeTo: node object to
        """

        self.__nodeFrom = nodeFrom
        self.__nodeTo = nodeTo
        self.__time = time
        self.__distance = distance

    def getTime(self):
        return int(self.__time)

    def setTime(self, time):
        self.__time = time

    def getDistance(self):
        return self.__distance

    def setDistance(self, distance):
        self.__distance = distance

    def getNodeFrom(self):
        return self.__nodeFrom

    def getNodeTo(self):
        return self.__nodeTo

